smart-contracts/contracts/storage/VideoStorage.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "../../node_modules/@openzeppelin/contracts/access/Ownable.sol";
+import "../lib/SafeMath64.sol";
+
+// Any change to this struct requires VideoStorage migration
+struct Video {
+  uint64 channelId;
+  bool isActive;
+  bool isExisting;
+}
+
+contract VideoStorage is Ownable {
+  using SafeMath64 for uint64;
+
+  mapping(uint64 => Video) private videoById;
+  mapping(uint64 => uint64) public videoCountByChannelId;
+  uint64 nextVideoId = 1;
```

Comment:
I'm used to 1-based index when it comes to "entity ids", since that's usually how ids are assigned in `MySQL`/`PostgresSQL` databases etc. (and since query node uses PostgresSQL database it seems to make sense).

Also it looks like currently we're using 1-based index in the runtime too: https://github.com/Joystream/joystream/blob/4865e9b3fc436cb0878778922bfc1cb3689d1acc/node/src/chain_spec/content_config.rs#L238